jpms-test/src/main/java/module-info.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Dummy module to prevent Maven Compiler Plugin from failing if {@code module-info.java} exists
+ * only in test sources:
+ *
+ * <blockquote>
+ *
+ * Can't compile test sources when main sources are missing a module descriptor
```

Comment:
Maybe there is another way to work around this; the Maven Compiler Plugin [checks if the main output directory exists](https://github.com/apache/maven-compiler-plugin/blob/maven-compiler-plugin-3.13.0/src/main/java/org/apache/maven/plugin/compiler/TestCompilerMojo.java#L364-L367), but maybe that check is a bit brittle?

https://lists.apache.org/thread/c3f71v8tf3643dto6bwf3ctcos4gt5cq suggests it might even depend on the JUnit version being used (?).

So for now having this dummy module declaration might be the most reliable workaround.